python/cudf/cudf/dataframe/series.py

```diff
@@ -361,66 +359,39 @@ def empty(self):
         return not len(self)
 
     def __getitem__(self, arg):
-        if isinstance(
-            arg, (list, np.ndarray, pd.Series, range, Index, DeviceNDArray)
-        ):
-            if len(arg) == 0:
-                arg = Series(np.array([], dtype="int32"))
-            else:
-                arg = Series(arg)
-        if isinstance(arg, Series):
-            if issubclass(arg.dtype.type, np.integer):
-                maps = columnops.as_column(arg).data.mem
-                index = self.index.take(maps)
-                selvals = self._column.take(maps)
-            elif arg.dtype in [np.bool, np.bool_]:
-                selvals = self._column.apply_boolean_mask(arg)
-                index = self.index.as_column().apply_boolean_mask(arg)
-            else:
-                raise NotImplementedError(arg.dtype)
-            return self._copy_construct(data=selvals, index=index)
-        elif isinstance(arg, slice):
-            index = self.index[arg]  # slice index
-            col = self._column[arg]  # slice column
-            return self._copy_construct(data=col, index=index)
-        elif isinstance(arg, Number):
-            # The following triggers a IndexError if out-of-bound
-            return self._column.element_indexing(arg)
-        else:
-            raise NotImplementedError(type(arg))
-
-    def take(self, indices, ignore_index=False):
-        """Return Series by taking values from the corresponding *indices*.
-        """
-        from cudf import Series
+        data = self._column[arg]
+        index = self.index.take(arg)
+        if utils.is_scalar(data) or data is None:
+            return data
+        return self._copy_construct(data=data, index=index)
 
-        if isinstance(indices, Series):
-            indices = indices.to_gpu_array()
+    def __setitem__(self, key, value):
+        # coerce value into a scalar or column
+        if utils.is_scalar(value):
+            value = utils.to_cudf_compatible_scalar(value)
         else:
-            indices = Buffer(indices).to_gpu_array()
-        # Handle zero size
-        if indices.size == 0:
-            return self._copy_construct(
-                data=self.data[:0], index=self.index[:0]
-            )
+            value = columnops.as_column(value)
 
-        if self.dtype == np.dtype("object"):
-            return self[indices]
+        if hasattr(value, "dtype") and pd.api.types.is_numeric_dtype(
+            value.dtype
+        ):
+            # normalize types if necessary:
+            if not isinstance(key, int):
```

Comment:
Not sure if you care about handling it, but this won't handle if `key` is a numpy scalar, i.e. `key = np.int32(1)`